### ISSUE TITLE
DISPLAY makes Agda loop

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#1644]
Attempting to fix printing with DISPLAY pragma makes Agda loop

``` agda
module _ where

module M where
  postulate A : Set

module N = M  -- This alias introduces a display form M.A --> N.A

open M

postulate
  a : A

{-# DISPLAY N.A = A #-} -- Makes Agda loop

test : Set
test = a

-- Without DISPLAY: N.A !=< Set of type Set
-- ERROR SHOULD BE:   A !=< Set of type Set
```

**Comment by Saizan** [comment]
If we go the way of what's planned for issue #892 and desugar "open M" to "module M' = M; open M'" then things work out fine.

**Comment by UlfNorell** [comment]
Closed by 6374fee. Now refuses to add the looping display form.
